lib/services/local/headlessauthn_watcher_test.go

```diff
@@ -106,6 +104,8 @@ func TestHeadlessAuthenticationWatcher(t *testing.T) {
 				Item: *item,
 			})
 
+			time.Sleep(100 * time.Millisecond)
```

Comment:
Could you add a comment saying why we need the sleep?

Reply:
I'll update this too use custom ticker logic instead of trying to rely on `require.Eventually`. This will make it more efficient and more clear.